markdown/developers/pipeline_structure.md

```diff
@@ -0,0 +1,74 @@
+
+## General
+
+The best way to write a new pipeline is to start from the template (see here). After doing that, you will have a number of files in your repository with different purposes. Here's what they are for:
+
+### Files
+
+* `main.nf`: This is the main nextflow file which will get executed if the pipeline is run. It parses all parameters, loads the modules with options and defines the specific order in which these are executed.
+
+* `nextflow.config`: The main nextflow configuration file. It contains the default pipeline parameters, nextflow configuration options and information like pipeline and nextflow version, among others.
+
+* `README.md`: Basic information about the pipeline and usage
+
+* `nextflow_json.schema`: The JSON schema file is used for pipeline parameter specification. It can be automatically created usint the `tools` package. In the pipeline, this is used for printing out a help message to the user or validating the input parameters. The nf-core tools utility package furthermore uses the JSON Schema for launching pipelines. Furthermore, `nextflow.config` defines different profiles that can be used to run the pipeline. These include the test profiles, which load, when used, the above mentioned configuration files into the `nextflow.config` file. Additionaly, the following profiles are available by default: `debug`, `conda`, `docker`, `singularity` and `podman`‚ which are used to run the pipeline in debug mode, using a conda environment or a using a container technology (docker, singularity or podman).
+
+* `CHANGELOG.md`: The change log contains information about, well, changes made to the pipeline.
+
+* `LICENSE`: The license - should be MIT
+
+* `CODE_OF_CONDUCT.md`: The nf-core code of conduct.
+
+* .gitattributes: Lists GitHub settings
+
+* .gitignore: Lists files that should be ignored by GitHub.
+
+### Directories
+
+* `.github`: Other GitHub specific files, e.g. for specifying templates and GitHub actions
+
+* `assets`: Any additional files needed for the pipeline
+
+* `bin`: The bin directory simply contains any scripts that must be directly accessible within a pipeline process. Anything in this directory can be directly called from within Nextflow processes.
+
+* `conf`: The `conf` directory contains a `base.config` file which is always loaded into `nextflow.config` and describes basic pipeline configurations, like CPU and memory usage for processes with low, medium and high requirements. Additionaly, `conf` contains a `igenomes.config` file which describes the locations of popular genomes that can be automatically downloaded for a pipeline run. Finally, the `test.config` and `test_full.config` files are test configurations that are loaded during test runs. Since DSL2, it also contains a `modules.config` file, which defines module-specific configurations and is explained further down in the "DSL2 and modules" section.
```

Comment:
```suggestion
* `conf/`: Configuration files, including a `base.config` file which is always loaded into `nextflow.config` and describes basic pipeline configurations, like CPU and memory usage for processes with low, medium and high requirements. Additionaly, most pipelines also have a `igenomes.config` file which describes the locations of popular genomes that can be automatically downloaded for a pipeline run. Finally, the `test.config` and `test_full.config` files are test configurations that are loaded during test runs. Since DSL2, it also contains a `modules.config` file, which defines module-specific configurations and is explained further down in the "DSL2 and modules" section.
```